cmake/SundialsSetupConfig.cmake

```
@@ -33,7 +33,7 @@ else()
 endif()
 
 # prepare substitution variable SUNDIALS_USE_GENERIC_MATH for sundials_config.h
-set(SUNDIALS_USE_GENERIC_MATH "SUNDIALS_MATH_LIBRARY")
+set(SUNDIALS_USE_GENERIC_MATH "${SUNDIALS_MATH_LIBRARY}")
```

To match the old behavior I think this should be
```suggestion
if(SUNDIALS_C_COMPILER_HAS_MATH_PRECISIONS)
  set(SUNDIALS_USE_GENERIC_MATH FALSE)
else()
  set(SUNDIALS_USE_GENERIC_MATH TRUE)
endif()
```

---

I don't think that is what we want, because `SUNDIALS_C_COMPILER_HAS_MATH_PRECISIONS` does not mean that `SUNDIALS_MATH_LIBRARY` was provided - it just means the compile check passed. `SUNDIALS_MATH_LIBRARY` not being provided is equivalent to `USE_GENERIC_MATH=OFF`. The way the code currently is, if `SUNDIALS_MATH_LIBRARY` is set, then `SUNDIALS_USE_GENERIC_MATH` will be defined. If it not set, than `SUNDIALS_USE_GENERIC_MATH` will not be defined. This matches the old behavior I think.

---

I guess it depends on which old behavior of `USE_GENERIC_MATH` you mean. I meant how it was used to indicate if `std-c math` functions were linked. Perhaps you mean how it was used to determine if the double precision functions were to be used?

---

I think either way we can make the case for it, but since it is more likely `SUNDIALS_USE_GENERIC_MATH` (if ever was used by anyone) was to determine which precision math functions were used, we can go with your suggestion.

---

Sorry for the confusion, yes I was thinking how this as indicating if the double precision functions were being used or not.